src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+                    else:
+                        scales = [max(e) for e in zip(scales, hdr.scales)]
+
+                    if offsets is None:
+                        offsets = hdr.offsets
+                    else:
+                        offsets = [min(e) for e in zip(offsets, hdr.offsets)]
+
+            return GeoMetadata(
+                paths=sources,
+                extents=BoundingBox(
+                    minx=mins[0],
+                    miny=mins[1],
+                    minz=mins[2],
+                    maxx=maxs[0],
+                    maxy=maxs[1],
+                    maxz=maxs[2],
+                ),
+                scales=tuple(scales),
+                offsets=tuple(offsets),
+            )
+
+
+def get_geometry_metadata(
+    sources: Iterable[os.PathLike],
+    config: Mapping[str, Any] = None,
+    verbose: bool = False,
+    id: str = "pointcloud_metadata",
+    trace: bool = False,
+    log_uri: Optional[str] = None,
+) -> GeoMetadata:
+    """Return geospatial metadata for a sequence of input geometry data files
+
+    :param sources: A sequence of paths or path to input
+    :param config: dict configuration to pass on tiledb.VFS
+    :param verbose: bool, enable verbose logging, default is False
+    :param trace: bool, enable trace logging, default is False
+    :param log_uri: Optional[str] = None,
+    :Return: GeoMetadata, a populated GeoMetadata object
+    """
+    # note this will be refactored to use /vsipyopener in fiona
+    if len(sources) == 0:
+        raise ValueError("Input point cloud datasets required")
+
+    with tiledb.scope_ctx(config):
+        tiledb.VFS()
+        logger = get_logger_wrapper(verbose)
+        with Profiler(array_uri=log_uri, id=id, trace=trace):
+            # for geometries we need the maximum extents
+            # and to check geometry types / crs are the same
+            schema = None
+            full_extents = None
+            crs = None
+
+            for pth in sources:
+                logger.debug("finding metadata for %r", pth)
+                with fiona.open(pth) as src:
+                    if crs is None:
+                        crs = src.crs
+                    else:
+                        if src.crs != crs:
+                            raise ValueError("All datasets need to be in the same CRS")
+
+                    if schema is None:
+                        schema = src.schema
+                    else:
+                        if src.schema != schema:
+                            raise ValueError(
+                                "All datasets need to have the same schema"
+                            )
+
+                    # 2.5 is supported internally but not needed for ingest
+                    extents = BoundingBox(
+                        minx=src.bounds[0],
+                        miny=src.bounds[1],
+                        maxx=src.bounds[2],
+                        maxy=src.bounds[3],
```

Comment:
likewise, this becomes `BoundingBox(min=Point.at(src.bounds[:2], max=src.bounds[2:])`

Reply:
as before, lets discuss, because we can use shapely if this is a must have change.